concourse/scripts/spec/tag-release_spec.rb

```diff
@@ -0,0 +1,87 @@
+require 'fileutils'
+
+prepare_aruba_config()
+
+RSpec.describe "tag-release.sh", :type => :aruba do
+  before(:each) do
+    prepare_ssh_keys()
+    config_git_email_and_name()
+    setup_cloned_repository()
+    clone_repository("paas-cf")
+
+    set_environment_variable 'GIT_REPO_URL', File.join(@workdir, "origin_repo")
+  end
+
+  context("when I checkout the before last tag of 'previous-*'") do
+    before(:each) do
+      bash_c("cd paas-cf && git checkout previous-0.0.2")
+    end
+
+
+    it("should promote the same tag version for 'next-*'") do
+      run("./tag-release.sh next- test_aws_account test_env \"previous-*\"")
+      expect(last_command_started).to have_exit_status(0)
+
+      bash_c_quiet("cd paas-cf && echo a && git tag -l \"next-*\" --sort version:refname | tail -n 1")
+      expect(last_command_started.output).to include("next-0.0.2")
+      expect(last_command_started.output).not_to include("next-0.0.3")
+    end
+
+    it("should have push the new tag to the remote repo") do
+      run("./tag-release.sh next- test_aws_account test_env \"previous-*\"")
+      expect(last_command_started).to have_exit_status(0)
+
+      clone_repository("paas-cf-new")
+
+      bash_c_quiet("cd paas-cf-new && git tag -l \"next-*\" --sort version:refname | tail -n 1")
+      expect(last_command_started).to have_output include_output_string "next-0.0.2"
+      expect(last_command_started).not_to have_output include_output_string "next-0.0.3"
+    end
+  end
+end
+
+def prepare_ssh_keys()
+    # The script expects git-keys/git-keys.tar.gz
+    bash_c(%q{
+      mkdir -p git-keys
+      cd git-keys
+      ssh-keygen -f git-key -N ""
+      tar -cvzf git-keys.tar.gz git-key git-key.pub
+   })
+end
+
+def config_git_email_and_name()
+    bash_c(%q{
+      set -e
+      git config --global user.email "you@example.com"
+      git config --global user.name "Your Name"
+    })
+end
+
+def setup_cloned_repository()
+    bash_c(%q{
+      mkdir origin_repo
+      cd origin_repo
+      git init .
+      git commit --allow-empty -m "First commit"
+
+      git tag previous-0.0.1
```

Comment:
I think the tag creation might belong in the individual `context` blocks rather than the outer `before` block, so that it's easier to understand what the individual dependencies of each test are (because I suspect they will diverge).